src/systems/filecoin_blockchain/storage_power_consensus/storage_power_actor.md

```diff
@@ -14,58 +14,69 @@ title: Storage Power Actor
 
-Put another way, power accounting in the SPC is delayed between storage being proven or faulted, and power being updated in the power table (and so for leader election). This ensures fairness over time.
+Sector ProveCommit is the first time power is proven to the network and hence power is first added through `_rtAddPowerForSector` at `OnSectorProveCommit`. Power is also added when a sector's TemporaryFault period has ended. Miners are expected to prove over all their sectors that contribute to their power. `_rtDeductClaimedPowerForSectorAssert` is called to decrement a miner's power. This is called when a sector expires or invoked by miner through `OnSectorTerminate` and when a sector enters TemporaryFault through `OnSectorTemporaryFaultEffectiveBegin`. Both `_rtAddPowerForSector` and `_rtDeductClaimedPowerForSectorAssert` are currently called at `OnSectorModifyWeightDesc` as power is determined by `SectorStorageWeightDesc` and `SectorStorageWeightDesc` is only modified when a miner calls `ExtendSectorExpiration` to extend a sector's duration. This may or may not have an impact on power but the machinery is in place to preserve the flexibility.
 
 The Miner lifecycle in the power table should be roughly as follows:
 
 - MinerRegistration: A new miner with an associated worker public key and address is registered on the power table by the storage mining subsystem, along with their associated sector size (there is only one per worker).
 - UpdatePower: These power increments and decrements are called by various storage actor (and must thus be verified by every full node on the network). Specifically:
-    - Power is incremented to account for a new SectorCommitment at the first PoSt past the first ProvingPeriod.
-    - All Power is decremented immediately after a missed PoSt.
-    - Power is decremented immediately after faults are declared, proportional to the faulty sector size.
-    - Power is incremented after a PoSt recovering from a fault.
-    - Power is definitively removed from the Power Table past the sector failure timeout (see {{<sref storage_faults>}})
+    - Power is incremented at SectorProveCommit
+    - All Power of a particular miner is decremented immediately after a missed SurprisePoSt (DetectedFault).
+    - A particular sector's power is decremented `OnSectorTemporaryFaultEffectiveBegin` when TemporaryFault is declared on the sector.
+    - A particular sector's power is added back `OnSectorTemporaryFaultEffectiveEnd` and miner is expected to prove over this sector. 
+    - A particular sector's power is removed when the sector is terminated through sector expiration or miner invocation (`_rtTerminateSector`).
 To summarize, only sectors in the Active state will command power. A Sector becomes Active after their first PoSt from Committed and Recovering stages. Power is immediately decremented when an Active Sector enters the Failing state (through DeclareFaults or Cron) and when an Active Sector expires.
```

Comment:
[blocking] Note that with ePoSt this is no longer necessary. Sectors can become active right after the commitment. No need to wait for the first PoSt anymore (since you can only profit from an active sector through ePoSt)

Reply:
Ack and it has been updated.